tap_hubspot/__init__.py

```diff
@@ -929,6 +934,51 @@ def sync_deal_pipelines(STATE, ctx):
     singer.write_state(STATE)
     return STATE
 
+def sync_contact_list(STATE, ctx):
+    bookmark_key = 'versionTimestamp'
+    stream_id = singer.get_currently_syncing(STATE)
+    stream_class = stream_id.split(":")[0]
+    list_id = stream_id.split(":")[1]
+    list_name = stream_id.split(":")[2]
+    
+    catalog = ctx.get_catalog_from_id(stream_id)
+    start = utils.strptime_with_tz(get_start(STATE, stream_id, bookmark_key))
+    LOGGER.info('sync_contact_list: %s from %s', list_name, bookmark_key)
+
+    max_bk_value = start
+    schema = load_schema(stream_class)
+
+    singer.write_schema(stream_id, schema, ["vid"], [bookmark_key], catalog.get('stream_alias'))
+
+    url = get_url("contact_list", list_id = list_id)
+
+    vids = []
+    with Transformer(UNIX_MILLISECONDS_INTEGER_DATETIME_PARSING) as bumble_bee:
+        for row in gen_request(STATE, stream_id, url, default_contact_params, 'contacts', 'has-more', ['vid-offset'], ['vidOffset']):
+            modified_time = None
+            if bookmark_key in row:
+                modified_time = utils.strptime_with_tz(
+                    _transform_datetime( # pylint: disable=protected-access
+                        row[bookmark_key],
+                        UNIX_MILLISECONDS_INTEGER_DATETIME_PARSING))
+
+            if not modified_time or modified_time >= start:
+                vids.append(row['vid'])
+
+            if modified_time and modified_time >= max_bk_value:
+                max_bk_value = modified_time
+
+            if len(vids) == 100:
```

Comment:
what is special of length 100? inside if block its calling same `_sync_contact_vids` with same param. Then it will again be called outside if block, why do we need to call the method twice?

Reply:
I took this part from how this `tap` imports contacts. It looks like it's batching requests to the HubSpot server taking contact details of up to 100 contacts at a time.
And after processing all the items in the contact list we call for details of the rest of unsynced contacts